### PR TITLE
*(ticdc): switch to unified DML library

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #3242 

### What is changed and how it works?

as title.

I'll file another PR to better represent omitted values when build RowChange. MySQL binlog also has `binlog_row_image` variable which can be supported as well.

### Check List 

Tests 

 - original unit test https://ci.pingcap.net/blue/organizations/jenkins/tiflow-verify-pipeline/detail/tiflow-verify-pipeline/1071/pipeline
 - original integration test https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_integration_test/detail/cdc_ghpr_integration_test/9618/pipeline

Code changes

Side effects

Related changes

### Release note 

```release-note 
`None`.
```

### CODE REVIEW

**Comment by lance6716** [comment]
CDC's integration test passed https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_integration_test/detail/cdc_ghpr_integration_test/9573/pipeline

I'll refine the code and unit tests soon

**Comment by lance6716** [comment]
/cc @liuzix @amyangfei @maxshuang @hi-rustin 

I'll check and remove some TODO comment soon and I think it's OK to review currently

File: cdc/entry/mounter.go [hunk 1]
```diff
@@ -31,11 +31,12 @@ import (
 	"github.com/pingcap/tidb/table"
 	"github.com/pingcap/tidb/tablecodec"
 	"github.com/pingcap/tidb/types"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
```
Comment (hi-rustin) [inline cdc/entry/mounter.go:36]:
```suggestion
```

File: cdc/model/sink.go [hunk 2]
```diff
@@ -21,9 +21,10 @@ import (
 
 	"github.com/pingcap/log"
 	"github.com/pingcap/tidb/parser/model"
+	"go.uber.org/zap"
+
```
Comment (hi-rustin) [inline cdc/model/sink.go:25]:
```suggestion
```

File: cdc/processor/pipeline/sorter.go [hunk 3]
```diff
@@ -21,6 +21,9 @@ import (
 	"github.com/pingcap/errors"
 	"github.com/pingcap/failpoint"
 	"github.com/pingcap/log"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
```
Comment (hi-rustin) [inline cdc/processor/pipeline/sorter.go:26]:
```suggestion
```

File: cdc/redo/reader/file.go [hunk 4]
```diff
@@ -30,13 +30,14 @@ import (
 	"github.com/pingcap/errors"
 	"github.com/pingcap/log"
 	"github.com/pingcap/tidb/br/pkg/storage"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
```
Comment (hi-rustin) [inline cdc/redo/reader/file.go:36]:
```suggestion
```

File: cdc/sink/mysql.go [hunk 5]
```diff
@@ -29,6 +29,9 @@ import (
 	"github.com/pingcap/log"
 	timodel "github.com/pingcap/tidb/parser/model"
 	"github.com/pingcap/tidb/parser/mysql"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/zap"
+
```
Comment (hi-rustin) [inline cdc/sink/mysql.go:34]:
```suggestion
```

File: cdc/entry/mounter.go (hunk 1 above)
Comment (lance6716) [reply cdc/entry/mounter.go:36]: I have configurated my IDE to separate 3rd-party imports with packages from itself. Should I turn if off?

File: cdc/sink/mysql.go [hunk 6]
```diff
@@ -598,6 +600,33 @@ func (s *mysqlSink) execDMLWithMaxRetries(ctx context.Context, dmls *preparedDML
 	}, retry.WithBackoffBaseDelay(backoffBaseDelayInMs), retry.WithBackoffMaxDelay(backoffMaxDelayInMs), retry.WithMaxTries(defaultDMLMaxRetryTime), retry.WithIsRetryableErr(isRetryableDMLError))
 }
 
+func convert2RowChange(row *model.RowChangedEvent) *sqlmodel.RowChange {
```
[the remark this answers is not in the document]
Comment (lance6716) [reply cdc/sink/mysql.go:605]: https://github.com/pingcap/tiflow/pull/4587/commits/4117e24048ab84b3e6576966306b0339086fc0f3

**Comment by lance6716** [comment]
Now I think in this PR it's too late to convert RowChangedEvent into the strcuture of library, at least it should be moved prior to causality. A good place maybe `dispatchAndExecTxns`. I'll work on it later.

**Comment by lance6716** [comment]
replaced by https://github.com/pingcap/tiflow/pull/7548

**Pull request closed without merging**